Some problems in the MacOS environment

OS:MacOS Monterey 12.0.1
# 1.When I build deviceShifu binaries directly execute this command, system report an error
> CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-socket deviceshifu/cmd/cmdSocket/main.go
> go build command-line-arguments: mkdir /output/: read-only file system

MacOS does not allow users to create folders in the root directory
# 2.When build deviceShifu docker images execute this command, It Seems that `$(go env GOOS)` And  `$(go env GOARCH)` are invalid
> make buildx-load-image-deviceshifu
>docker buildx build --platform= / -f /Users/xxx/Desktop/Project/Go/shifu/deviceshifu/Dockerfile.deviceshifu --build-arg PROJECT_ROOT="/Users/xxx/Desktop/Project/Go/shifu" /Users/xxx/Desktop/Project/Go/shifu -t edgehub/deviceshifu-http-http:v0.0.1 --load


Use ` $(shell go env GOOS) `And `$(shell go env GOARCH)` replace them)

# 3. After `Question2`, when executing the dockerfile,appear another error 
> make buildx-load-image-deviceshifu
> ...
>  => ERROR exporting to image
> ------
> exporting to image:
> ------
> error: failed to solve: operating system is not supported
> make: *** [buildx-load-image-deviceshifu] Error 1

It Seems That `Darwin` is not supported in this images `edgehub/distroless-static:nonroot` 

On `macOS` GOOS should be `linux` instead of `darwin`
# 4.`guide-on-writing-an-application-for-deviceShifu.md` documentation error

for example 
> ./test/scripts/shifu-application-demo-env-setup.sh apply applicationDemo     

the `shifu-application-demo-env-setup.sh` not exists. This markdown should be rewrite

--- a/Makefile
+++ b/Makefile
@@ -12,18 +12,18 @@ buildx-push-image-deviceshifu:
 	docker buildx build --platform=linux/amd64,linux/arm64,linux/arm -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuSocket --build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} -t edgehub/deviceshifu-http-socket:${IMAGE_VERSION} --push
 
 buildx-load-image-deviceshifu:
-	docker buildx build --platform=$(go env GOOS)/$(go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifu \
+	docker buildx build --platform=linux/$(shell go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifu \
 		--build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} \
 		-t edgehub/deviceshifu-http-http:${IMAGE_VERSION} --load
-	docker buildx build --platform=$(go env GOOS)/$(go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuMQTT \
+	docker buildx build --platform=linux/$(shell go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuMQTT \
 	 	--build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} \
 		-t edgehub/deviceshifu-http-mqtt:${IMAGE_VERSION} --load
-	docker buildx build --platform=$(go env GOOS)/$(go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuSocket \
-		--build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} 
-		-t edgehub/deviceshifu-http-socket:${IMAGE_VERSION} --load	
-	docker buildx build --platform=$(go env GOOS)/$(go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuOPCUA \
-		--build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} 
-		-t edgehub/deviceshifu-http-opcua:${IMAGE_VERSION} --load	
+	docker buildx build --platform=linux/$(shell go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuSocket \
+		--build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} \
+		-t edgehub/deviceshifu-http-socket:${IMAGE_VERSION} --load
+	docker buildx build --platform=linux/$(shell go env GOARCH) -f ${PROJECT_ROOT}/deviceshifu/Dockerfile.deviceshifuOPCUA \
+		--build-arg PROJECT_ROOT="${PROJECT_ROOT}" ${PROJECT_ROOT} \
+		-t edgehub/deviceshifu-http-opcua:${IMAGE_VERSION} --load
 
 .PHONY: download-demo-files
 download-demo-files:
@@ -101,4 +101,4 @@ docker-push-image-mockdevices:
 
 .PHONY: clean-images
 clean-images:
-	docker rmi $(sudo docker images | grep 'edgehub')
+	docker rmi $(shell sudo docker images | grep 'edgehub')

--- a/docs/design/design-shifu-zh.md
+++ b/docs/design/design-shifu-zh.md
@@ -134,7 +134,7 @@ Shifu 可以运行在所有主要平台上，包括但不限于 x86/64, ARM64 �
 
 ***deviceShifu*** 有以下两种运行方式：
 1. ***standalone mode***: ***standalone mode*** 是被设计为用来管理单个复杂的 ***edgeDevice*** 比如机械手臂来提供高质量的一对一 ***edgeDevice*** 管理
-2. ***swarm mode***: ***swarm mode*** 是被设计用来管理多个简单且同类的 ***edgeDevice*** 比如温度计来提供搞笑的一对多 ***edgeDevice*** 管理
+2. ***swarm mode***: ***swarm mode*** 是被设计用来管理多个简单且同类的 ***edgeDevice*** 比如温度计来提供高效的一对多 ***edgeDevice*** 管理
 
 ### 架构图
 

--- a/docs/guide/development/build-shifu-zh.md
+++ b/docs/guide/development/build-shifu-zh.md
@@ -11,19 +11,19 @@
 
 `HTTP 转 HTTP` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-http deviceshifu/cmd/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-http deviceshifu/cmd/main.go
 ```
 `HTTP 转 Socket` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-socket deviceshifu/cmd/cmdSocket/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-socket deviceshifu/cmd/cmdSocket/main.go
 ```
 `HTTP 转 MQTT` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-mqtt deviceshifu/cmd/cmdMQTT/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-mqtt deviceshifu/cmd/cmdMQTT/main.go
 ```
 `HTTP 转 OPC UA` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-opcua deviceshifu/cmd/cmdOPCUA/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-opcua deviceshifu/cmd/cmdOPCUA/main.go
 ```
 
 ### 2. 构建 ***deviceShifu*** `Docker` 镜像:

--- a/docs/guide/development/build-shifu.md
+++ b/docs/guide/development/build-shifu.md
@@ -11,19 +11,19 @@ Navigate to `Shifu`'s root directory, use the following commands to build differ
 
 For `HTTP to HTTP` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-http deviceshifu/cmd/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-http deviceshifu/cmd/main.go
 ```
 For `HTTP to Socket` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-socket deviceshifu/cmd/cmdSocket/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-socket deviceshifu/cmd/cmdSocket/main.go
 ```
 For `HTTP to MQTT` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-mqtt deviceshifu/cmd/cmdMQTT/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-mqtt deviceshifu/cmd/cmdMQTT/main.go
 ```
 For `HTTP to OPC UA` ***deviceShifu***:
 ```sh
-CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o /output/deviceshifu-http-opcua deviceshifu/cmd/cmdOPCUA/main.go
+CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -a -o output/deviceshifu-http-opcua deviceshifu/cmd/cmdOPCUA/main.go
 ```
 
 ### 2. Build ***deviceShifu*** `Docker` images:

--- a/docs/guide/guide-on-writing-an-application-for-deviceShifu-zh.md
+++ b/docs/guide/guide-on-writing-an-application-for-deviceShifu-zh.md
@@ -14,7 +14,7 @@
 在 `shifu` 根目录下，运行下面两条命令来运行 *shifu* 和演示温度计的 *deviceShifu*：
 
 ```
-./test/scripts/shifu-application-demo-env-setup.sh apply applicationDemo                       # setup and start shifu services for this demo
+./test/scripts/deviceshifu-setup.sh apply applicationDemo          # setup and start shifu services for this demo
 ./test/scripts/deviceshifu-demo.sh apply edgedevice-thermometer    # connect fake thermometer to shifu
 ```
 ### 2. 温度检测程序
@@ -119,4 +119,5 @@ kubectl logs -n default high-temperature-detector -f
 2021/10/18 10:35:49 High temperature: 30
 2021/10/18 10:35:51 High temperature: 30
 2021/10/18 10:35:53 Low temperature: 15
+
 

--- a/docs/guide/guide-on-writing-an-application-for-deviceShifu.md
+++ b/docs/guide/guide-on-writing-an-application-for-deviceShifu.md
@@ -10,9 +10,9 @@ The following example requires [Go](https://golang.org/dl/), [Docker](https://do
 ### 1. Start *Shifu* and connect a simple thermometer device
 The deployment config for a fake thermometer which produces a integer value representing the temperature read should be already in the `shifu/deviceshifu/examples/demo_device` directory.\
 The device driver has an API `read_value` which returns such integer value.
-Under `shifu` root directory, we can run the following three commands to have *shifu* and the fake thermometer *deviceShifu* ready:
+Under `shifu` root directory, we can run the following two commands to have *shifu* and the fake thermometer *deviceShifu* ready:
 ```
-./test/scripts/shifu-application-demo-env-setup.sh apply applicationDemo                       # setup and start shifu services for this demo
+./test/scripts/deviceshifu-setup.sh apply applicationDemo          # setup and start shifu services for this demo
 ./test/scripts/deviceshifu-demo.sh apply edgedevice-thermometer    # connect fake thermometer to shifu
 ```
 ### 2. High temperature detector application

